.gitlab-ci.yml

```diff
@@ -169,6 +169,38 @@ deploy_to_reliability_env:
   rules:
     - when: never # dd-trace-dotnet does not use reliability environment
 
+download-serverless-artifacts:
+  stage: package
+  image: registry.ddbuild.io/docker:20.10.13-gbi-focal
+  tags: [ "arch:amd64" ]
+  needs: []
+  rules:
+    - if: $DOTNET_PACKAGE_VERSION
+      when: on_success
+    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(-prerelease)?$/' # Manually triggered as artifacts are from the Github release
+      when: manual
+      allow_failure: false
```

Comment:
I'm not entirely sure how to think about this 🤔 _Should_ we be manually triggering this when we do a release? Because it doesn't actually do anything special for releases, right (in contrast to the SSI artifacts where we take a different path for releases).

If it's not doing anything different, should we just _skip_ for the tag instead? I'm not entirely sure the best option 🤔

Reply:
So, the release is not really automated from the .NET repo side, but on a downstream on our side. To not cause any accidental releases for layers.

This `manual` trigger is more if someone wants to release a testing AWS Lambda layer for .NET, which is why I decided to have it manual. As it'd be triggering our downstream pipeline on every push for every branch

Reply:
> But we do need it for the tag, since we want to make sure we can get the artifacts directly from the specific tag, which then will be used from a manual trigger downstream.

The thing is, what does "directly from the specific tag" _mean_ 😅 It's "special" like this for the download-ssi-artifacts because we use the artifacts directly from the GitHub release. But we're not doing anything like that here. It's not using the tag for anything in your script, so it's _not_ getting "the artifacts directly from the specific tag". It's always using the COMMIT_SHA I believe. Which is not a big problem I think, it just reiterates that the "tag" logic here is superfluous I think 😃 

It's not a big deal, as it all depends on downstream processes tbh, but was just flagging it in general, because it _may_ cause issues for you down the line 🙂